### PR TITLE
move mamba installation

### PR DESCRIPTION
**Pull request by akoumpa**
# What does this PR do ?

Should save 20-30mins in CI docker image builds.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Review by thomasdhc — DISMISSED** [review]
LGTM

**Review by JRD971000 — DISMISSED** [review]
LGTM!

**Review by ko3n1g — CHANGES_REQUESTED** [review]
Git’s `--branch` option only allows for branches or tags, but some some deps (especially Mcore), we want to checkout commits (for the nightly automated bump). You can see the failure in the log https://github.com/NVIDIA/NeMo/actions/runs/10888545114/job/30221221383?pr=10447#step:4:19640 

git official documentation: https://git-scm.com/docs/git-clone

**Comment by akoumpa** [comment]
Thanks @ko3n1g , I've reverted the git clone changes for megatron.

File: Dockerfile.ci [hunk 1]
```diff
@@ -31,9 +31,29 @@ EOF
 
 WORKDIR /workspace
 
+# Install Mamba
+ARG MAMBA_TAG=v2.0.3
+ARG CAUSAL_CONV_TAG=v1.2.2.post1
+
+RUN <<"EOF" bash -ex
+# Mamba dependancy installation
+git clone --depth 1 --branch ${MAMBA_TAG} https://github.com/state-spaces/mamba.git && \
```
Comment (ko3n1g) [inline Dockerfile.ci:40]: Out of curiosity: why not a pip+git install instead? Are there benefits to this approach? No need to change it if results in the same

**Review by ko3n1g — APPROVED** [review]
LGTM. Just one question b/c I‘m curious and let’s definitely wait for the CI result

File: Dockerfile.ci (hunk 1 above)
Comment (akoumpa) [reply Dockerfile.ci:40]: I think it picks the package and we want to build it from source, I'm not sure how to trigger the same setup.py command via pip.

**Pull request merged**